File: Infrastructure/Data/IBookRepo.cs
using EntityModels;
using System.Collections.Generic;

namespace Infrastructure.Data
{
    public interface IBookRepo : ICRUD<BookEM, long>
    {
        void UpdateAuthors(long bookId, IEnumerable<long> authorIds);
        DataTableResponseEM<BookEM> GetList(DataTableRequestEM model);
        IEnumerable<BookEM> GetALL();
    }
}

File: InfrastructureVB/Data/IBookRepo.vb
Imports EntityModels

Namespace Data
    Public Interface IBookRepo
        Inherits ICRUD(Of BookEM, Long)

        Sub UpdateAuthors(bookId As Long, authorIds As IEnumerable(Of Long))
        Function GetList(model As DataTableRequestEM) As DataTableResponseEM(Of BookEM)

    End Interface
End Namespace